radix-engine/src/blueprints/resource/auth_zone.rs

```diff
@@ -291,4 +292,43 @@ impl AuthZoneBlueprint {
 
         Ok(IndexedScryptoValue::from_typed(&proofs))
     }
+
+    pub(crate) fn drop<Y>(
+        input: &IndexedScryptoValue,
+        api: &mut Y,
+    ) -> Result<IndexedScryptoValue, RuntimeError>
+    where
+        Y: KernelSubstateApi<SystemLockData> + ClientApi<RuntimeError>,
+    {
+        // TODO: add `drop` callback for drop atomicity, which will remove the necessity of kernel api.
+
+        let input: AuthZoneDropInput = input.as_typed().map_err(|e| {
+            RuntimeError::SystemUpstreamError(SystemUpstreamError::InputDecodeError(e))
+        })?;
+
+        // Detach proofs from the auth zone
+        let handle = api.kernel_lock_substate(
```

Comment:
Instead of locking, you could just use the substates returned from `drop_object` no?

Reply:
Commented in another thread.